[CI] Netty4WriteThrottlingHandlerTests testThrottleLargeCompositeMessage failing

### CI Link

https://gradle-enterprise.elastic.co/s/jsxipcu4fdofg/tests/task/:modules:transport-netty4:test/details/org.elasticsearch.transport.netty4.Netty4WriteThrottlingHandlerTests/testThrottleLargeCompositeMessage?top-execution=1

### Repro line

./gradlew ":modules:transport-netty4:test" --tests "org.elasticsearch.transport.netty4.Netty4WriteThrottlingHandlerTests.testThrottleLargeCompositeMessage" -Dtests.seed=4F09C67841D50E8C -Dtests.locale=zh-HK -Dtests.timezone=Portugal -Druntime.java=23	

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

```REPRODUCE WITH: ./gradlew ":modules:transport-netty4:test" --tests "org.elasticsearch.transport.netty4.Netty4WriteThrottlingHandlerTests.testThrottleLargeCompositeMessage" -Dtests.seed=4F09C67841D50E8C -Dtests.locale=zh-HK -Dtests.timezone=Portugal -Druntime.java=23

Netty4WriteThrottlingHandlerTests > testThrottleLargeCompositeMessage FAILED
    java.lang.AssertionError: 
    Expected: a collection with size one of {<9>, <10>}
         but: collection size was <11>
        at __randomizedtesting.SeedInfo.seed([4F09C67841D50E8C:51F6A42ACBCF3D34]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
        at org.elasticsearch.test.ESTestCase.assertThat(ESTestCase.java:2625)
        at org.elasticsearch.transport.netty4.Netty4WriteThrottlingHandlerTests.testThrottleLargeCompositeMessage(Netty4WriteThrottlingHandlerTests.java:123)```

--- a/modules/transport-netty4/src/test/java/org/elasticsearch/transport/netty4/Netty4WriteThrottlingHandlerTests.java
+++ b/modules/transport-netty4/src/test/java/org/elasticsearch/transport/netty4/Netty4WriteThrottlingHandlerTests.java
@@ -109,6 +109,7 @@ public class Netty4WriteThrottlingHandlerTests extends ESTestCase {
             Netty4WriteThrottlingHandler.MAX_BYTES_PER_WRITE * fullSizeChunks + extraChunkSize
         );
         int splitOffset = randomIntBetween(0, messageBytes.length);
+        int lastChunkSizeOfTheFirstSplit = splitOffset % Netty4WriteThrottlingHandler.MAX_BYTES_PER_WRITE;
         final BytesReference message = CompositeBytesReference.of(
             new BytesArray(messageBytes, 0, splitOffset),
             new BytesArray(messageBytes, splitOffset, messageBytes.length - splitOffset)
@@ -120,7 +121,9 @@ public class Netty4WriteThrottlingHandlerTests extends ESTestCase {
         assertFalse(promise.isDone());
         embeddedChannel.flush();
         assertTrue(promise.isDone());
-        assertThat(seen, hasSize(oneOf(fullSizeChunks, fullSizeChunks + 1)));
+        // If the extra chunk size is greater than the last chunk size for the first half of the split, it means we will need to send
+        // (extraChunkSize - lastChunkSizeOfTheFirstSplit) bytes as the very last chunk of the entire message.
+        assertThat(seen, hasSize(oneOf(fullSizeChunks, fullSizeChunks + 1 + (extraChunkSize > lastChunkSizeOfTheFirstSplit ? 1 : 0))));
         assertTrue(capturingHandler.didWriteAfterThrottled);
         assertBufferEquals(Unpooled.compositeBuffer().addComponents(true, seen), message);
     }

